chore(lint): clear the 2 react-hooks/exhaustive-deps warnings, then drop --max-warnings to 0

`npm run lint` in `frontend/` reports 2 warnings. They pass only because the script allows `--max-warnings 5`.

```
frontend/src/admin/AdminPanel.jsx
  221:9  warning  The 'tabs' array makes the dependencies of useEffect Hook (at line 248)
                  change on every render. To fix this, wrap the initialization of 'tabs'
                  in its own useMemo() Hook   react-hooks/exhaustive-deps

frontend/src/admin/UserManagement.jsx
  111:6  warning  React Hook useEffect has a missing dependency: 'fetchUsers'.
                  Either include it or remove the dependency array   react-hooks/exhaustive-deps
```

Both are real: `AdminPanel.jsx:221` rebuilds `tabs` every render, so the effect at `:248` re-runs every render; `UserManagement.jsx:111` has a stale-closure risk on `fetchUsers`.

The backend (`npm run lint` at root) is already at **0 problems**.

## Fix

1. Wrap `tabs` in `useMemo` with its real dependencies.
2. Either add `fetchUsers` to the dep array (wrapping it in `useCallback`) or restructure so it isn't needed.
3. Once clear, **drop `--max-warnings 5` to `0`** in `frontend/package.json`'s lint script, so the next warning fails the gate instead of hiding under the allowance.

Step 3 is the point of the issue — the threshold is why these two sat unnoticed.

## Verification

`make gate` exits 0 with the stricter threshold in place.

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -14,7 +14,7 @@
     "psi": "node scripts/run-psi.mjs",
     "psi:dev": "npm run psi -- --url https://dev.settimesdotca.pages.dev",
     "psi:prod": "npm run psi -- --url https://settimes.ca",
-    "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 5",
+    "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0",
     "lint:fix": "eslint . --ext js,jsx --fix",
     "format": "prettier --write \"src/**/*.{js,jsx,json,css}\"",
     "format:check": "prettier --check \"src/**/*.{js,jsx,json,css}\"",

--- a/frontend/src/admin/AdminPanel.jsx
+++ b/frontend/src/admin/AdminPanel.jsx
@@ -1,4 +1,4 @@
-import { useState, useEffect, useCallback } from 'react'
+import { useState, useEffect, useCallback, useMemo } from 'react'
 import { eventsApi } from '../utils/adminApi'
 import EventsTab from './EventsTab'
 import VenuesTab from './VenuesTab'
@@ -217,17 +217,22 @@ export default function AdminPanel({ currentUser, onLogout }) {
 
   const selectedEvent = events.find(e => e.id === selectedEventId)
 
-  // Dynamic Tabs Configuration
-  const tabs = [
-    { id: 'events', label: 'Events' },
-    // Only show Lineup if an event is selected
-    ...(selectedEventId ? [{ id: 'lineup', label: 'Lineup' }] : []),
-    { id: 'roster', label: 'Roster' },
-    { id: 'venues', label: 'Venues' },
-    ...(canManageUsers ? [{ id: 'users', label: 'Users' }] : []),
-    { id: 'settings', label: 'Settings' },
-    ...(isAdmin ? [{ id: 'platform', label: 'Platform' }] : []),
-  ]
+  // Dynamic Tabs Configuration. Memoized because the effect below depends on
+  // `tabs`: rebuilt inline it is a fresh array every render, so the effect
+  // re-ran on every render instead of only when the tab set actually changes.
+  const tabs = useMemo(
+    () => [
+      { id: 'events', label: 'Events' },
+      // Only show Lineup if an event is selected
+      ...(selectedEventId ? [{ id: 'lineup', label: 'Lineup' }] : []),
+      { id: 'roster', label: 'Roster' },
+      { id: 'venues', label: 'Venues' },
+      ...(canManageUsers ? [{ id: 'users', label: 'Users' }] : []),
+      { id: 'settings', label: 'Settings' },
+      ...(isAdmin ? [{ id: 'platform', label: 'Platform' }] : []),
+    ],
+    [selectedEventId, canManageUsers, isAdmin]
+  )
 
   useEffect(() => {
     if (tabs.some(tab => tab.id === activeTab)) {

--- a/frontend/src/admin/UserManagement.jsx
+++ b/frontend/src/admin/UserManagement.jsx
@@ -1,4 +1,4 @@
-import { useState, useEffect, useMemo } from 'react'
+import { useState, useEffect, useMemo, useCallback } from 'react'
 import { Key, Pencil, Plus, Trash, UserCheck, UserX } from 'lucide-react'
 import RoleBadge from './components/RoleBadge'
 import UserFormModal from './components/UserFormModal'
@@ -94,7 +94,10 @@ export default function UserManagement({ showToast }) {
     }))
   }
 
-  const fetchUsers = async () => {
+  // useCallback so the mount effect can declare it as a dependency without
+  // re-running every render. `showToast` is itself useCallback-wrapped by
+  // AdminPanel, so this identity is stable.
+  const fetchUsers = useCallback(async () => {
     try {
       const data = await usersApi.getAll()
       setUsers(data.users)
@@ -104,11 +107,11 @@ export default function UserManagement({ showToast }) {
     } finally {
       setLoading(false)
     }
-  }
+  }, [showToast])
 
   useEffect(() => {
     fetchUsers()
-  }, [])
+  }, [fetchUsers])
 
   const handleCreateUser = async userData => {
     setActionLoading(true)

